corehq/motech/repeaters/views.py

```diff
@@ -135,6 +135,21 @@ def page_url(self):
         return reverse(AddFormRepeaterView.urlname, args=[self.domain])
 
 
+def get_edit_repeater_view(request, domain, repeater_id):
+    from corehq.motech.openmrs.views import EditOpenmrsRepeaterView
+
+    try:
+        repeater = Repeater.get(repeater_id)
+    except ResourceNotFound:
+        raise Http404()
+    view_class = {
+        'CaseRepeater': EditCaseRepeaterView,
+        'FormRepeater': EditFormRepeaterView,
+        'OpenmrsRepeater': EditOpenmrsRepeaterView,
+    }.get(repeater.doc_type, EditRepeaterView)
+    return view_class.as_view()(request, domain, repeater_id=repeater_id, repeater_type=repeater.doc_type)
```

Comment:
Hmmm. The more I ponder this line, the less happy I am with it. I don't think it's *wrong*, but I do think it breaks the Django paradigm, because it's doing routing in views.py instead of in urls.py.